package.json

```diff
@@ -15,13 +15,14 @@
     "test:only": "mocha",
     "test:leak": "node --expose-gc test/leak/index.js",
     "test:quick": "rollup -c && mocha -b",
+    "test:typescript": "mv src src_tmp; tsc -p test/typescript; mv src_tmp src",
     "pretest-coverage": "npm run build",
     "test-coverage": "rm -rf coverage/* && istanbul cover --report json node_modules/.bin/_mocha -- -u exports -R spec test/test.js",
     "posttest-coverage": "remap-istanbul -i coverage/coverage-final.json -o coverage/coverage-remapped.json -b dist && remap-istanbul -i coverage/coverage-final.json -o coverage/coverage-remapped.lcov -t lcovonly -b dist && remap-istanbul -i coverage/coverage-final.json -o coverage/coverage-remapped -t html -b dist",
     "ci": "npm run test-coverage && codecov < coverage/coverage-remapped.lcov",
     "build": "git rev-parse HEAD > .commithash && rollup -c && tsc -p tsconfig-types.json && chmod a+x bin/rollup",
     "watch": "rollup -cw",
-    "prepublishOnly": "npm run lint && npm run test:only && npm run test:leak",
+    "prepublishOnly": "npm run lint && npm run test:only && npm run test:leak && npm run test:typescript",
```

Comment:
Answering your question when this test should run: As it appears, at least on my system, that this test is rather fast, I think it would be ok to include it in the test script. Otherwise the person publishing it would  be the only one fixing issues here (i.e. probably me).
However I would add it after the regular test, i.e. `npm run test:only && npm run test:typescript`. Adding it to `test` will also make it run on CI. The `ci` script, on the other hand, seems to be unused at the moment.

Another issue I stumble upon is that the typings in the `dist` folder are never cleaned up which resulted in a lot of weird errors for me when running this test. My suggestion is to remove the `dist/typings` folder (or maybe even the `dist` folder) as part of the `build` script.

Reply:
Fixed in 65a713d. I'm not a windows guy so I'm not sure whether commands like `rm` or `mv` are still a cross-platform problem these days. Since `rm -rf` is already used in the `test-coverage`script I used the same in `prebuild`. If that's a concern we could use `rimraf` instead. I'm not sure about the `mv` though. Maybe [`cash-mv`](https://www.npmjs.com/package/cash-mv) is an option?

Reply:
Alright, appveyor just answered my question. Will fix...